cpp/src/arrow/CMakeLists.txt

```diff
@@ -132,10 +132,13 @@ set(ARROW_SRCS
     util/task-group.cc
     util/thread-pool.cc
     util/trie.cc
-    util/uri.cc
     util/utf8.cc
     vendored/datetime/tz.cpp)
 
+if(ARROW_WITH_URIPARSER)
+  set(ARROW_SRCS ${ARROW_SRCS} util/uri.cc)
+endif()
```

Comment:
uriparser is not a large dependency -- making it optional could grow onerous as time goes on, but we can see

Reply:
Where is the Uri Parser being used currently? I couldn't find any usage.